packages/@aws-cdk/aws-iam/test/role.from-role-arn.test.ts

```diff
@@ -0,0 +1,533 @@
+            beforeEach(() => {
+              policyStack = new Stack(app, 'PolicyStack', { env: { account: roleAccount } });
+              importedRole.attachInlinePolicy(somePolicy(policyStack, 'MyPolicy'));
+            });
+
+            test("correctly attaches the Policy to the imported role", () => {
+              assertRoleHasAttachedPolicy(policyStack, roleName, 'MyPolicy');
+            });
+          });
+
+          describe('a different account than in the ARN of the imported role', () => {
+            beforeEach(() => {
+              policyStack = new Stack(app, 'PolicyStack', { env: { account: notRoleAccount } });
+              importedRole.attachInlinePolicy(somePolicy(policyStack, 'MyPolicy'));
+            });
+
+            test("does NOT attach the Policy to the imported role", () => {
+              assertPolicyDidNotAttachToRole(policyStack, 'MyPolicy');
+            });
+          });
+        });
+      });
+    });
+
+    describe('into a stack with account set to', () => {
+      describe('the same account as in the ARN the role was imported with', () => {
+        beforeEach(() => {
+          roleStack = new Stack(app, 'RoleStack', { env: { account: roleAccount } });
+          importedRole = Role.fromRoleArn(roleStack, 'ImportedRole',
+            `arn:aws:iam::${roleAccount}:role/${roleName}`);
+        });
+
+        describe('then adding a PolicyStatement to it', () => {
+          let addToPolicyResult: boolean;
+
+          beforeEach(() => {
+            addToPolicyResult = importedRole.addToPolicy(somePolicyStatement());
+          });
+
+          test('returns true', () => {
+            expect(addToPolicyResult).toBe(true);
+          });
+
+          test("generates a default Policy resource pointing at the imported role's physical name", () => {
+            assertRoleHasDefaultPolicy(roleStack, roleName);
+          });
+        });
+
+        describe('then attaching a Policy to it', () => {
+          describe('that belongs to the same stack as the imported role', () => {
+            beforeEach(() => {
+              importedRole.attachInlinePolicy(somePolicy(roleStack, 'MyPolicy'));
+            });
+
+            test("correctly attaches the Policy to the imported role", () => {
+              assertRoleHasAttachedPolicy(roleStack, roleName, 'MyPolicy');
+            });
+          });
+
+          describe('that belongs to an env-agnostic stack', () => {
+            let policyStack: Stack;
+
+            beforeEach(() => {
+              policyStack = new Stack(app, 'PolicyStack');
+              importedRole.attachInlinePolicy(somePolicy(policyStack, 'MyPolicy'));
+            });
+
+            test("correctly attaches the Policy to the imported role", () => {
+              assertRoleHasAttachedPolicy(policyStack, roleName, 'MyPolicy');
+            });
+          });
+
+          describe('that belongs to a non-env-agnostic stack, with account set to', () => {
```

Comment:
Instead of the double negative here, maybe let's call it a "targeted stack"? Very minor point.

Reply:
I've never heard us using the word "targeted" to mean "a stack with a specific account and/or region set", but sure, changed.

Reply:
You're right. We're starting today! 😛